gobblin-runtime-hadoop/src/main/java/gobblin/runtime/instance/plugin/hadoop/HadoopKerberosKeytabAuthenticationPlugin.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+package gobblin.runtime.instance.plugin.hadoop;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.security.UserGroupInformation;
+
+import com.typesafe.config.Config;
+
+import gobblin.annotation.Alias;
+import gobblin.runtime.api.GobblinInstanceDriver;
+import gobblin.runtime.instance.hadoop.HadoopConfigLoader;
+import gobblin.runtime.instance.plugin.BaseIdlePluginImpl;
+
+/**
+ * Loads a Kerberos keytab file for Hadoop authentication.
+ */
+@Alias(value="hadoopLoginFromKeytab")
+public class HadoopKerberosKeytabAuthenticationPlugin extends BaseIdlePluginImpl {
+  public static final String INSTANCE_CONFIG_PREFIX = "gobblin.instance";
+  public static final String CONFIG_PREFIX = INSTANCE_CONFIG_PREFIX + ".hadoop";
+  public static final String LOGIN_USER_KEY = "loginUser";
+  public static final String LOGIN_USER_KEYTAB_FILE_KEY = "loginUserKeytabFile";
+  public static final String LOGIN_USER_FULL_KEY = CONFIG_PREFIX + "." + LOGIN_USER_KEY;
+  public static final String LOGIN_USER_KEYTAB_FILE_FULL_KEY =
+      CONFIG_PREFIX + "." + LOGIN_USER_KEYTAB_FILE_KEY;
+
+  private final String _loginUser;
+  private final String _loginUserKeytabFile;
+  private final Configuration _hadoopConf;
+
+  public HadoopKerberosKeytabAuthenticationPlugin(GobblinInstanceDriver instance) {
```

Comment:
I'm not sure this will fully do the keytab authentication. I believe you then have to set configuration variables. See https://github.com/linkedin/gobblin/blob/master/gobblin-azkaban/src/main/java/gobblin/azkaban/AzkabanJobLauncher.java#L133 for an example.